nsv13/code/modules/overmap/weapons/delta_overmap_ship_weapons/overmap_ship_weapon.dm

```diff
@@ -0,0 +1,383 @@
+			controller_strings += "Autonomous"
+		else
+			controller_strings += "AMS Control"
+	control_flag_string = controller_strings.Join(", ")
+	cached_control_flags = weapon_control_flags
+	return control_flag_string
+
+/**
+ * Called when AI tries to use a weapon when out of ammo. Attempts to initiate the rearm process.
+ * * Base implementation is generic, but you can override this if you want to do nonstandard rearms for a weapon :)
+ */
+/datum/overmap_ship_weapon/proc/try_initiating_resupply()
+	var/rearm_type
+	if(used_nonphysical_ammo == OSW_AMMO_LIGHT)
+		rearm_type = SHIP_RESUPPLYING_LIGHT
+	else
+		rearm_type = SHIP_RESUPPLYING_HEAVY
+
+	if(linked_overmap.ai_resupply_scheduled & rearm_type)
+		return
+	linked_overmap.ai_resupply_scheduled |= rearm_type
+
+	if(rearm_type == SHIP_RESUPPLYING_LIGHT)
+		addtimer(CALLBACK(linked_overmap, TYPE_PROC_REF(/obj/structure/overmap, ai_self_resupply_light)), linked_overmap.ai_light_resupply_time)
+	else
+		addtimer(CALLBACK(linked_overmap, TYPE_PROC_REF(/obj/structure/overmap, ai_self_resupply)), linked_overmap.ai_resupply_time)
+
+
+
+
+//Overmap object procs
+
+/**
+ * Recalculates the role specific weapon lists.
+ */
+/obj/structure/overmap/proc/recalc_role_weapon_lists()
+	if(length(pilot_weapon_datums))
+		pilot_weapon_datums.Cut()
+	if(length(gunner_weapon_datums))
+		gunner_weapon_datums.Cut()
+	if(length(autonomous_weapon_datums))
+		autonomous_weapon_datums.Cut()
+	if(length(pilotgunner_weapon_datums))
+		pilotgunner_weapon_datums.Cut()
+
+	for(var/datum/overmap_ship_weapon/osw as() in overmap_weapon_datums)
+		if(osw.weapon_control_flags & OSW_CONTROL_PILOT)
+			pilot_weapon_datums += osw
+		if(osw.weapon_control_flags & OSW_CONTROL_GUNNER)
+			gunner_weapon_datums += osw
+		if(osw.weapon_control_flags & OSW_CONTROL_AUTONOMOUS)
+			autonomous_weapon_datums += osw
+	pilotgunner_weapon_datums |= gunner_weapon_datums //Do not add datums twice.
+	pilotgunner_weapon_datums |= pilot_weapon_datums //Pilot guns will always end up ordered after gunner ones regardless of priority.
+
+/**
+ * Deletes all linked weapon datums.
+ */
+/obj/structure/overmap/proc/purge_overmap_weapon_datums()
+	for(var/datum/overmap_ship_weapon/osw as() in overmap_weapon_datums)
+		qdel(osw) //Qdel handles unlink here.
+
+/**
+ * Admin proc (do not use in the code)
+ * * Creates a new ship weapon datum of the passed type, and automatically links it to this ship.
+ * * weapon_type determines type of weapon created.
+ * * update_role_weapon_lists = FALSE prevents updating the weapon lists. Can be useful if you want to add several weapons before updating.
+ * * Does not have safeties. Make sure the type you hand it is actually a weapon or you'll get a free runtime.
+ */
+/obj/structure/overmap/proc/create_weapon_datum(weapon_type, update_role_weapon_lists = TRUE)
+	var/datum/overmap_ship_weapon/new_weapon = new weapon_type(src, update_role_weapon_lists)
+	return new_weapon
+
```

Comment:
Please rename the files to remove overmap_ship_weapon from all but this one.

Reply:
done